tempesta_fw/http.c

```diff
@@ -1997,22 +2007,24 @@ tfw_http_set_hdr_connection(TfwHttpMsg *hm, int conn_flg)
  * Add/Replace/Remove Keep-Alive header field to/from HTTP message.
  */
 static int
-tfw_http_set_hdr_keep_alive(TfwHttpMsg *hm, int conn_flg)
+tfw_http_set_hdr_keep_alive(TfwHttpMsg *hm, unsigned long conn_flg)
 {
 	int r;
 
-	if ((hm->flags & __TFW_HTTP_MSG_M_CONN_MASK) == conn_flg)
+	BUILD_BUG_ON(__TFW_HTTP_MSG_M_CONN_MASK > BITS_PER_LONG);
+	if ((hm->flags[0] & __TFW_HTTP_MSG_M_CONN_MASK) == conn_flg)
```

Comment:
This looks like a rather hacky way of doing this check. This code implicitly assumes that the connection's keep-alive/close flags are located somewhere in the first bits of the `hm->flags` bitmap (first 64 bits, in fact). That may be true, but it creates an unnecessary dependency on this implicit piece of knowledge. Again, there's the issue of whether these two flags are maintained as mutually exclusive or not, or whether these two flags  can be both set at the same time by the parser.

It appears that, according to changes in other parts of this patch, `conn_flag` is now guaranteed to only have one bit set, either `TFW_HTTP_CONN_KA` or `TFW_HTTP_CONN_CLOSE`. Then, perhaps, only a check for a specific flag is needed, like `test_bit(conn_flag, hm->flags)`.

Reply:
Uh, well, this `BUILD_BUG_ON()` states that all flags in the mask must be within the first 5 bits, assuming that `BITS_PER_LONG` is 64. Or, if there's just one flag and it's the 6-th bit. Was that the intention? It's non-obvious I would say.

Now that `conn_flag` is set to one specific bit/flag, Perhaps it's cleaner and clearer to just check for that bit/flag.